vspace/src/main/java/edu/asu/diging/vspace/web/staff/StaffSpacesSearchController.java

```diff
@@ -21,27 +21,41 @@
 @Controller
 public class StaffSpacesSearchController {
 
+    private static final int PAGE_SIZE = 5;
+
     @Autowired
     private ISpaceManager spaceManager;
 
     @RequestMapping("/staff/spaces/search")
-    public ResponseEntity <String> search(@RequestParam(value = "term", required = false) String search) {
-        List <ISpace> spaces = null;
+    public ResponseEntity<String> search(
+            @RequestParam(value = "term", required = false) String search,
+            @RequestParam(value = "page", required = false, defaultValue = "1") int page) {
+
+        List<ISpace> spaces;
+        boolean hasMore;
+
         if (search != null && !search.trim().isEmpty()) {
-            spaces = spaceManager.findByName(search);
+            spaces = spaceManager.findByNamePaginated(search, page, PAGE_SIZE);
+            hasMore = spaces.size() == PAGE_SIZE;
```

Comment:
That's not necessarily true, is it? If each page has 10 entries (10 being the number of items on a page), then this would yield true for every page if there are 100 items, even the last one, wouldn't it?